midend/parserUnroll.cpp

```diff
@@ -178,13 +178,22 @@ class ParserStateRewriter : public Transform {
     inline size_t getIndex() { return currentIndex; }
 
  protected:
-    const IR::Type* getTypeArray(const IR::Node* element) {
+const IR::Type* getTypeArray(const IR::Node* element) {
         if (element->is<IR::ArrayIndex>()) {
             const IR::Expression* left = element->to<IR::ArrayIndex>()->left;
             if (left->type->is<IR::Type_Stack>())
                 return left->type->to<IR::Type_Stack>()->elementType;
         }
-        return typeMap->getType(element, true);
+        auto* currentType = typeMap->getType(element);
```

Comment:
How can it happen that type information is missing here? Is this a problem with type inference?

Reply:
The problem with right part of the [assignment](https://github.com/VolodymyrPeschanenkoIntel/p4c/blob/205467c5841284ee7af871bd0198b803f1e04c06/testdata/p4_16_samples/parser-unroll-test7.p4#L37). If element is hdr.h[0].i1.index then  typeMap->getType(element) returns nullptr.